flow-client/src/main/java/com/vaadin/client/flow/util/ClientJsonCodec.java

```
@@ -61,12 +61,22 @@ private ClientJsonCodec() {
      */
     public static StateNode decodeStateNode(StateTree tree, JsonValue json) {
         if (json.getType() == JsonType.OBJECT) {
-            // Check for @v object format
-            String type = getStringProperty(json, "@v");
-            if ("node".equals(type)) {
-                int nodeId = (int) getNumberProperty(json, "id");
+            // Check for compact @v-node format
+            JsonValue nodeIdValue = getProperty(json, "@v-node");
+            if (nodeIdValue != null && nodeIdValue.getType() == JsonType.NUMBER) {
```

Not sure if its relevant but it might be more resilient to raise an error rather than just ignoring if the value is not a number